src/uu/du/src/du.rs

```diff
@@ -360,12 +446,18 @@ fn du(
                                     depth + 1,
                                     seen_inodes,
                                     exclude,
-                                ));
+                                    tx,
+                                )?);
                             } else {
                                 my_stat.size += this_stat.size;
                                 my_stat.blocks += this_stat.blocks;
                                 my_stat.inodes += 1;
                                 if options.all {
+                                    tx.send(StatPrintInfo {
+                                        stat: this_stat.clone(),
```

Comment:
is the clone necessary ?

Reply:
I was able to avoid cloning by making some pretty significant updates to the `du` function. I'm having it pass ownership of `Stat`s to the printing thread as soon as it's done using them, instead of collecting them into a vector and returning the contents of the vector in an iterator.

It seems like the only purpose of collecting the stats and returning them in an iterator was so the caller of `du` could print the results, so I think this new approach still preserves the original intention. 

I still have a good bit of work to do before this PR is ready, but could you let me know if my updates to the `du` function are a reasonable direction to take things, or if there's some reason this isn't a good approach? Thanks!